nucypher/blockchain/eth/sol/source/contracts/staking_contracts/PoolingStakingContract.sol

```diff
@@ -0,0 +1,260 @@
+pragma solidity ^0.6.1;
+
+
+import "zeppelin/ownership/Ownable.sol";
+import "zeppelin/math/SafeMath.sol";
+import "contracts/staking_contracts/AbstractStakingContract.sol";
+
+
+/**
+* @notice Contract acts as delegate for sub-stakers and owner
+**/
+contract PoolingStakingContract is AbstractStakingContract, Ownable {
+    using SafeMath for uint256;
+
+    event TokensDeposited(address indexed sender, uint256 value, uint256 depositedTokens);
+    event TokensWithdrawn(address indexed sender, uint256 value, uint256 depositedTokens);
+    event ETHWithdrawn(address indexed sender, uint256 value);
+    event DepositSet(address indexed sender, bool value);
+
+    struct Delegator {
+        uint256 depositedTokens;
+        uint256 withdrawnReward;
+        uint256 withdrawnETH;
+    }
+
+    StakingEscrow public immutable escrow;
+
+    uint256 public totalDepositedTokens;
```

Comment:
I think we should change this approach. Instead of assuming we're tracking here deposited tokens in absolute numbers, we can follow an approach where users get a share (i.e. percentage) of the pool. The cool thing about this is that we could consider tokenize participation in these pools, similarly to what DeFi projects are currently doing. That means that delegators receive an amount of the pool's own token (let's call it `pNU` for the sake of the argument), in exchange for their `NU`. This would drastically improve the viability of staking for smaller participants, as it will provide them of quick entry and exit (e.g, by buying/selling `pNU` in Uniswap).